src/ArrayInterface.jl

```diff
@@ -36,8 +36,26 @@ const MatAdjTrans{T,M<:AbstractMatrix{T}} = Union{Transpose{T,M},Adjoint{T,M}}
 const UpTri{T,M} = Union{UpperTriangular{T,M},UnitUpperTriangular{T,M}}
 const LoTri{T,M} = Union{LowerTriangular{T,M},UnitLowerTriangular{T,M}}
 
-@inline static_length(a::UnitRange{T}) where {T} = last(a) - first(a) + oneunit(T)
-@inline static_length(x) = Static.maybe_static(known_length, length, x)
+"""
+    length(A) -> Union{Int,StaticInt}
+
+Returns the length of `A`.  If the length is known at compile time, it is
+returned as `Static` number.  Otherwise, `ArrayInterface.length(A)` is identical
+to `Base.length(A)`.
+
+```julia
+julia> using StaticArrays, ArrayInterface
+
+julia> A = @SMatrix rand(3,4);
+
+julia> ArrayInterface.length(A)
+static(12)
+```
+"""
+@inline length(a::UnitRange{T}) where {T} = last(a) - first(a) + oneunit(T)
```

Comment:
One last question before I push a revision. I am wondering why there is a specific method for `UnitRange` here.  I tried removing it and everything seemed to be fine.

Reply:
@chriselrod added that a while back when the underlying call in base was overly complicated and slow. I'm pretty sure that has changed since. However, `UnitRange` can't ever have a static length so I'd leave that to Chris to change if it's no longer necessary.